lib/msal-common/src/client/AuthorizationCodeClient.ts

```diff
@@ -72,6 +75,17 @@ export class AuthorizationCodeClient extends BaseClient {
 
         // Retrieve requestId from response headers
         const requestId = response.headers?.[HeaderNames.X_MS_REQUEST_ID];
+        const httpVer = response.headers?.[HeaderNames.X_MS_HTTP_VERSION];
+        atsMeasurement?.addStaticFields({
+            httpVer: httpVer
+        });
+        // cache http version header
+        this.logger.verbose("Cache http version header for auth code ");
+        // CHECK IF cache manager is of type browser cache manager
+        if("setItem" in this.cacheManager){
```

Comment:
hmm not sure I like this...Why do we need to cache the http metadata?

Reply:
1) for instance discovery call(login.microsoftonline.com):- store the http version 
2) for /token:- check the http version, if different from the instance discovery call, override it. 

Storing it to retain the version across these network calls, and sometimes the http version is not available for /token(due to not being included in ACCESS_CONTROL_EXPOSE_HEADERS). (More:- https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Expose-Headers) 
This saved creating the submeasurements solely for the purpose of logging http version, e.g- for redirect flow. 



Alternatively, the http version can be logged as : http_ver_ token (for /token) and http_ver_metadata(for instance discovery metadata call) if cache usage is not desirable/ has performance concerns.